cdap-cli/src/main/java/co/cask/cdap/cli/command/CreateNamespaceCommand.java

```diff
@@ -45,22 +45,38 @@ public CreateNamespaceCommand(CLIConfig cliConfig, NamespaceClient namespaceClie
   @Override
   public void perform(Arguments arguments, PrintStream output) throws Exception {
     String name = arguments.get(ArgumentName.NAMESPACE_NAME.toString());
-    String description = arguments.get(ArgumentName.NAMESPACE_DESCRIPTION.toString(), "");
+
+    String description = arguments.getOptional(ArgumentName.NAMESPACE_DESCRIPTION.toString(), "");
+    String principal = arguments.getOptional(ArgumentName.NAMESPACE_PRINCIPAL.toString(), null);
+    String keytabPath = arguments.getOptional(ArgumentName.NAMESPACE_KEYTAB_PATH.toString(), null);
+    String hbaseNamespace = arguments.getOptional(ArgumentName.NAMESPACE_HBASE_NAMESPACE.toString(), null);
+    String hiveDatabase = arguments.getOptional(ArgumentName.NAMESPACE_HIVE_DATABASE.toString(), null);
+    String schedulerQueueName = arguments.getOptional(ArgumentName.NAMESPACE_SCHEDULER_QUEUENAME.toString(), "");
```

Comment:
we can still default this to null, and it will be set to empty string when build() is called below